### PR TITLE
WIFI-1105: Bug fix for survey report per radio

### PR DESCRIPTION
**Pull request by NagConnectus**
package:opensync

Signed-off-by: Nagendrababu <nagendrababu.bonkuri@connectus.ai>

### CODE REVIEW

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c [hunk 1]
```diff
@@ -192,43 +192,81 @@ void target_survey_record_free(target_survey_record_t *result)
...
 			survey = NULL;
 		}
 	}
-
 	(*survey_cb)(survey_list, survey_ctx, ret);
-
 	return ret;
 }
 
+
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c:269]: White space change unnecessary.

File: feeds/wlan-ap/opensync/src/vendor/tip/src/lib/target/inc/target_tip.h [hunk 2]
```diff
@@ -52,6 +52,7 @@ typedef struct
 	uint32_t chan_tx;
 	uint32_t chan_noise;
 	uint32_t duration_ms;
+	uint32_t chan_use;
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/vendor/tip/src/lib/target/inc/target_tip.h:55]: 'chan_use' is not a great name.  'chan_in_use' or 'active_chan' would have been more readable.

**Review by RickSommerville — COMMENTED** [review]
Mostly picking at inconsequential items.  Looks good

File: feeds/wlan-ap/opensync/src/vendor/tip/src/lib/target/inc/target_tip.h (hunk 2 above)
Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/vendor/tip/src/lib/target/inc/target_tip.h:55]: done

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c (hunk 1 above)
Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c:269]: done

**Review by RickSommerville — APPROVED** [review]
Looks good.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c [hunk 3]
```diff
@@ -192,40 +192,77 @@ void target_survey_record_free(target_survey_record_t *result)
...
 	while (!ds_dlist_is_empty(&raw_survey_list)) {
 		survey = ds_dlist_head(&raw_survey_list);
 		ds_dlist_remove(&raw_survey_list, survey);
 		LOGI("Survey entry dur %d busy %d tx %d rx %d chan %d active %d type %d",
-			survey->duration_ms, survey->chan_busy, survey->chan_tx, survey->chan_rx, survey->info.chan, survey->chan_active, scan_type);
-		if ((scan_type == RADIO_SCAN_TYPE_ONCHAN) && (survey->info.chan == chan_list[0])) {
-                	ds_dlist_insert_tail(survey_list, survey);
+				survey->duration_ms, survey->chan_busy, survey->chan_tx, survey->chan_rx,
+				survey->info.chan, survey->chan_active, scan_type );
+		if (scan_type == RADIO_SCAN_TYPE_ONCHAN) {
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c:248]: Don't need samples of duration_ms == 0  to be saved here.  Just add same test from line 254.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c [hunk 4]
```diff
@@ -192,40 +192,77 @@ void target_survey_record_free(target_survey_record_t *result)
...
+		while (!ds_dlist_is_empty(offchan_list)) {
+			survey = ds_dlist_head(offchan_list);
+			ds_dlist_remove(offchan_list, survey);
+			ds_dlist_insert_tail(survey_list, survey);
+		}
+	}
 	while (!ds_dlist_is_empty(&raw_survey_list)) {
 		survey = ds_dlist_head(&raw_survey_list);
 		ds_dlist_remove(&raw_survey_list, survey);
 		LOGI("Survey entry dur %d busy %d tx %d rx %d chan %d active %d type %d",
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c:245]: Change this to LOGD

**Review by RickSommerville — CHANGES_REQUESTED** [review]

**Comment by RickSommerville** [comment]
Also in target_stats_survey_convert() can you change the LOGI to LOGD.

**Review by RickSommerville — APPROVED** [review]

**Pull request merged**